src/global/reco/ParticleFlow_factory.h

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) Derek Anderson
+
+#pragma once
+
+#include <map>
+#include <vector>
+#include <utility>
+#include <spdlog/logger.h>
+#include <extensions/spdlog/SpdlogMixin.h>
+#include <extensions/jana/JChainMultifactoryT.h>
+// event data model definitions
+#include <edm4eic/ReconstructedParticleCollection.h>
+// for geometry services
+#include <services/geometry/dd4hep/DD4hep_service.h>
+// necessary algorithms and configurations
+#include "algorithms/reco/ParticleFlow.h"
+#include "algorithms/reco/ParticleFlowConfig.h"
+
+namespace eicrecon {
+
+  class ParticleFlow_factory :
+          public JChainMultifactoryT<ParticleFlowConfig>,
+          public SpdlogMixin {
+
+    public:
+
+      // aliases for brevity
+      using TrkInput     = std::pair<const edm4eic::ReconstructedParticleCollection*, const edm4eic::TrackSegmentCollection*>;
+      using CaloInput    = std::pair<const edm4eic::ClusterCollection*, const edm4eic::ClusterCollection*>;
+      using CaloIDs      = std::pair<uint32_t, uint32_t>;
+      using VecCaloInput = std::vector<CaloInput>;
+      using VecCaloIDs   = std::vector<CaloIDs>;
```

Comment:
These types will have to be kept in sync with the ParticleFlow.h header. You include that header here (you have to) so reuse the types from that header instead of redefining them.

But... kinda irrelevant comment since this all should go into the algorithm anyway :-)